Source/SIMPLib/Filtering/FilterPipeline.cpp

```diff
@@ -872,34 +867,28 @@ DataContainerArray::Pointer FilterPipeline::execute()
 
     // Emit that the filter is completed for those objects that care, even the disabled ones.
     emit filt->filterCompleted(filt.get());
+
+    notifyProgressMessage("", "", static_cast<int>(static_cast<float>(filtIndex+1) / (m_Pipeline.size()) * 100.0f));
   }
 
   disconnectSignalsSlots();
 
-  PipelineMessage completeMessage("", "", 0, PipelineMessage::MessageType::StatusMessage, -1);
   if(m_State == FilterPipeline::State::Canceling)
   {
-    completeMessage.setText("Pipeline Canceled");
     m_ExecutionResult = FilterPipeline::ExecutionResult::Canceled;
+    notifyStatusMessage("", "Pipeline Canceled");
   }
   else if(m_State == FilterPipeline::State::Executing)
   {
-    completeMessage.setText("Pipeline Complete");
     m_ExecutionResult = FilterPipeline::ExecutionResult::Completed;
+    notifyStatusMessage("", "Pipeline Complete");
   }
   else
   {
     // This should never get here
-    completeMessage.setText("Unsupported Pipeline Execution Result");
-    emit pipelineGeneratedMessage(completeMessage);
-
-    setErrorCondition(-210);
-    completeMessage.setType(PipelineMessage::MessageType::Error);
-    completeMessage.setCode(getErrorCondition());
+    notifyErrorMessage("", "Unsupported Pipeline Execution Result", -210);
```

Comment:
Enum checks should use switch statements.  They're generally more readable and shorter.  And if you end up where you shouldn't be, a good switch statement makes it very obvious where that is, unlike `else`.

Also, remove the `// This should never get here` comment.  We don't handle FilterPipeline error codes, so this is going to silently fail or worse.  If we absolutely shouldn't be there, throw an exception or in some way, shape, or form, make it obvious that we've gone off the rails.  The last time I hit a comment like that (and it was in this class), it was tracking down a bug where the GUI was being permanently locked but nothing was causing a message to be displayed or a crash where the debugger could say where we went off into undefined territory.  If you shouldn't be here, make it obvious and very loud when you do.